<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-03-24" />

<title>ihd subsample result</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">coxPHSuSIE-data-application</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ihd subsample result</h1>
<h4 class="date">2024-03-24</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-24
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>coxPHSuSIE-data-application/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240324code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240324)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240324code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240324)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215coxPHSuSIEdataapplicationtree068b07e7b89bbe3c2918f504cc8ce3fb3407aeb8targetblank068b07ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application/tree/068b07e7b89bbe3c2918f504cc8ce3fb3407aeb8" target="_blank">068b07e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215coxPHSuSIEdataapplicationtree068b07e7b89bbe3c2918f504cc8ce3fb3407aeb8targetblank068b07ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application/tree/068b07e7b89bbe3c2918f504cc8ce3fb3407aeb8" target="_blank">068b07e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/ihd_subsample.Rmd</code>) and
HTML (<code>docs/ihd_subsample.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application/blob/068b07e7b89bbe3c2918f504cc8ce3fb3407aeb8/analysis/ihd_subsample.Rmd" target="_blank">068b07e</a>
</td>
<td>
yunqi yang
</td>
<td>
2024-03-24
</td>
<td>
wflow_publish("analysis/ihd_subsample.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/yunqiyang0215/coxPHSuSIE-data-application/4127d18fa86f156df0f2d019253c40fbf456cbdb/docs/ihd_subsample.html" target="_blank">4127d18</a>
</td>
<td>
yunqi yang
</td>
<td>
2024-03-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application/blob/0e6158d8d354da6ae2110d8810c58b41200e02ca/analysis/ihd_subsample.Rmd" target="_blank">0e6158d</a>
</td>
<td>
yunqi yang
</td>
<td>
2024-03-24
</td>
<td>
wflow_publish("analysis/ihd_subsample.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="description" class="section level2">
<h2>Description:</h2>
<p>CoxPH-SuSIE result for Ischaemic Heart disease using subset of UKB
data. n.case and n.control = 10000.</p>
<pre class="r"><code>library(survival)
library(susieR)
devtools::load_all(&quot;/project2/mstephens/yunqiyang/surv-susie/logisticsusie&quot;)</code></pre>
<pre><code>ℹ Loading logisticsusie</code></pre>
<pre class="r"><code>res6 = readRDS(&quot;./data/ihd_subsample/fit.susie.chr6.rds&quot;)
res9 = readRDS(&quot;./data/ihd_subsample/fit.susie.chr9.rds&quot;)
snp_list = read.csv(&quot;./data/ihd_subsample/ihd_chr6.csv&quot;)</code></pre>
<div id="regions-on-chr6" class="section level3">
<h3>1. Regions on CHR6</h3>
<pre class="r"><code>fit = res6[[1]]
X = res6[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 0.30714700 0.27433395 0.22934896 0.22311034 0.20014679 0.19212692
 [7] 0.11486359 0.09563394 0.06034514 0.05729752</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
[1] 2889 2917 2932 2944

$cs$L2
 [1] 2010 2399 2563 2623 2634 2722 2752 2776 2796 2822 2852 2954


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.9926063     0.9952472       0.9952952
L2    0.7768360     0.9414890       0.9532242

$cs_index
[1] 1 2

$coverage
[1] 0.9998132 0.9710800

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>iter = fit$iter
iter</code></pre>
<pre><code>[1] 11</code></pre>
<pre class="r"><code>par(mfrow = c(1,2))
snps1 = colnames(X)[cs$cs$L1]
snp_names1 = unlist(lapply(snps1, function(x) unlist(strsplit(x, &quot;_&quot;))[1]))
colors &lt;- ifelse(snp_list$Variant %in% snp_names1, &quot;red&quot;, &quot;black&quot;)
plot(-log10(snp_list$p.value), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;)

snps2 = colnames(X)[cs$cs$L2]
snp_names2 = unlist(lapply(snps2, function(x) unlist(strsplit(x, &quot;_&quot;))[1]))
colors &lt;- ifelse(snp_list$Variant %in% snp_names2, &quot;red&quot;, &quot;black&quot;)
plot(-log10(snp_list$p.value), col = colors, xlab = &quot;SNP&quot;, ylab = &quot;-log10(p-value)&quot;)</code></pre>
<p><img src="figure/ihd_subsample.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/yunqiyang0215/coxPHSuSIE-data-application/blob/4127d18fa86f156df0f2d019253c40fbf456cbdb/docs/figure/ihd_subsample.Rmd/unnamed-chunk-5-1.png" target="_blank">4127d18</a>
</td>
<td>
yunqi yang
</td>
<td>
2024-03-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>snps_all = c(snps1, snps2)
snp_summary = cbind(snp_list[snp_list$Variant %in% c(snp_names1, snp_names2), ], pip[c(cs$cs$L1, cs$cs$L2)])
snp_summary</code></pre>
<pre><code>         Variant  Position Eff..Allele                            Trait
2609  rs10080815 160687412           T I20-I25 Ischaemic heart diseases
3052   rs9295127 160751123           A I20-I25 Ischaemic heart diseases
3282   rs1510226 160816409           C I20-I25 Ischaemic heart diseases
3372   rs9457925 160848743           G I20-I25 Ischaemic heart diseases
3388   rs4709431 160853746           C I20-I25 Ischaemic heart diseases
3517   rs1510224 160894878           C I20-I25 Ischaemic heart diseases
3558   rs9457927 160910282           G I20-I25 Ischaemic heart diseases
3584 rs117791490 160916058           T I20-I25 Ischaemic heart diseases
3609 rs117733303 160922870           A I20-I25 Ischaemic heart diseases
3649   rs7452960 160941641           A I20-I25 Ischaemic heart diseases
3697   rs3798220 160961137           T I20-I25 Ischaemic heart diseases
3756 rs118039278 160985526           A I20-I25 Ischaemic heart diseases
3793  rs74617384 160997118           A I20-I25 Ischaemic heart diseases
3813  rs55730499 161005610           T I20-I25 Ischaemic heart diseases
3832  rs10455872 161010118           A I20-I25 Ischaemic heart diseases
3846 rs140570886 161013013           C I20-I25 Ischaemic heart diseases
          Beta    p.value       MAF     HWE imp..score ORbeta.
2609 -0.023536 6.5029e-37 0.0197700 0.29280  genotyped   0.709
3052  0.025083 2.3237e-41 0.0197601 0.14700   0.988625   1.440
3282  0.025768 5.7231e-43 0.0194606 0.19590    0.98939   1.460
3372 -0.027202 7.9723e-46 0.0185700 0.02894  genotyped   0.672
3388  0.027201 5.8936e-46 0.0185813 0.42920    0.99832   1.490
3517  0.027400 3.1425e-46 0.0184464 0.48200   0.996268   1.490
3558  0.027591 2.6490e-46 0.0182479 0.36880   0.993092   1.500
3584  0.027433 2.3670e-46 0.0184267 0.45390   0.998037   1.490
3609 -0.027258 5.9794e-46 0.0185000 0.32660  genotyped   0.671
3649  0.027336 1.0542e-45 0.0182788 0.53910   0.998494   1.490
3697 -0.027400 6.4488e-46 0.0182900 0.39550  genotyped   0.670
3756  0.018096 1.1460e-79 0.0796169 1.00000   0.991241   1.300
3793 -0.017984 2.0848e-77 0.0781900 0.44650  genotyped   0.769
3813  0.018098 7.1470e-81 0.0804036 0.79310   0.997046   1.300
3832 -0.018141 1.0933e-80 0.0796800 0.85440  genotyped   0.767
3846  0.031895 4.3790e-51 0.0155470 0.14530    0.96051   1.590
     pip[c(cs$cs$L1, cs$cs$L2)]
2609                 0.19212692
3052                 0.30714700
3282                 0.22934896
3372                 0.27433395
3388                 0.02493094
3517                 0.02663487
3558                 0.02984840
3584                 0.20014679
3609                 0.22311034
3649                 0.11486359
3697                 0.04188253
3756                 0.09563394
3793                 0.05584116
3813                 0.05729752
3832                 0.06034514
3846                 0.05371242</code></pre>
</div>
<div id="regions-on-chr9" class="section level3">
<h3>2. Regions on CHR9</h3>
<pre class="r"><code>fit = res9[[1]]
X = res9[[2]]

pip &lt;- logisticsusie:::get_pip(fit$alpha)
effect_estimate &lt;- colSums(fit$alpha * fit$mu)
pip.sorted = sort(pip, decreasing = TRUE)
pip.sorted[1:10]</code></pre>
<pre><code> [1] 0.10909545 0.07408269 0.07267668 0.06485401 0.06482019 0.05852464
 [7] 0.05306919 0.04298331 0.04125483 0.04102229</code></pre>
<pre class="r"><code>class(fit) = &quot;susie&quot;
cs &lt;- susie_get_cs(fit, X)
cs</code></pre>
<pre><code>$cs
$cs$L1
 [1]  53  66  69  72  73  74  76  77  78  79  80  81  84  85  87  89  91  94  95
[20]  97  98 102 103 108 109 110 111 112 113 114 115 116 117 119


$purity
   min.abs.corr mean.abs.corr median.abs.corr
L1    0.8375944     0.9516436        0.955728

$cs_index
[1] 1

$coverage
[1] 0.9517232

$requested_coverage
[1] 0.95</code></pre>
<pre class="r"><code>iter = fit$iter
iter</code></pre>
<pre><code>[1] 11</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8 LC_NUMERIC=C         LC_TIME=C           
 [4] LC_COLLATE=C         LC_MONETARY=C        LC_MESSAGES=C       
 [7] LC_PAPER=C           LC_NAME=C            LC_ADDRESS=C        
[10] LC_TELEPHONE=C       LC_MEASUREMENT=C     LC_IDENTIFICATION=C 

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] logisticsusie_0.0.0.9004 testthat_3.1.4           susieR_0.12.35          
[4] survival_3.3-1           workflowr_1.7.0         

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8.3       lattice_0.20-45    prettyunits_1.1.1  getPass_0.2-2     
 [5] ps_1.7.0           assertthat_0.2.1   rprojroot_2.0.3    digest_0.6.29     
 [9] utf8_1.2.2         R6_2.5.1           plyr_1.8.7         evaluate_0.15     
[13] highr_0.9          httr_1.4.3         ggplot2_3.3.6      pillar_1.7.0      
[17] rlang_1.0.2        rstudioapi_0.13    irlba_2.3.5        whisker_0.4       
[21] callr_3.7.3        jquerylib_0.1.4    Matrix_1.5-3       rmarkdown_2.14    
[25] desc_1.4.1         devtools_2.4.3     splines_4.2.0      stringr_1.4.0     
[29] munsell_0.5.0      mixsqp_0.3-48      compiler_4.2.0     httpuv_1.6.5      
[33] xfun_0.30          pkgconfig_2.0.3    pkgbuild_1.3.1     htmltools_0.5.2   
[37] tidyselect_1.1.2   tibble_3.1.7       matrixStats_0.62.0 reshape_0.8.9     
[41] fansi_1.0.3        crayon_1.5.1       dplyr_1.0.9        withr_2.5.0       
[45] later_1.3.0        brio_1.1.3         grid_4.2.0         jsonlite_1.8.0    
[49] gtable_0.3.0       lifecycle_1.0.1    DBI_1.1.2          git2r_0.30.1      
[53] magrittr_2.0.3     scales_1.2.0       cli_3.3.0          stringi_1.7.6     
[57] cachem_1.0.6       remotes_2.4.2.1    fs_1.5.2           promises_1.2.0.1  
[61] bslib_0.3.1        ellipsis_0.3.2     generics_0.1.2     vctrs_0.4.1       
[65] tools_4.2.0        glue_1.6.2         purrr_0.3.4        processx_3.8.0    
[69] pkgload_1.2.4      fastmap_1.1.0      yaml_2.3.5         colorspace_2.0-3  
[73] sessioninfo_1.2.2  memoise_2.0.1      knitr_1.39         usethis_2.1.5     
[77] sass_0.4.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
